### ISSUE TITLE
Style/EmptyLines cop flags offences for empty lines used to group methods

### ISSUE THREAD

**Issue by hjdivad** [bbatsov/rubocop#2633]
The ruby style guide section on [empty lines](https://github.com/bbatsov/ruby-style-guide#empty-lines-between-methods)  states:

> Use empty lines between method definitions and also to break up methods into logical paragraphs internally.

It is common to *also* group methods within a class logically, as well as to group lines within a method internally.

However the `Style/EmptyLines` cop flags as offences methods with two blank lines between them, making it impossible to group the methods themselves.

For example, grouping readers and writers separately:
```rb
class Account
  def balance
    # compute balance
  end

  def name
    # return account name
  end


  def deposit!(amount:)
    # deposit amount
  end
end
```

Is there any interest in a PR that makes the [line offset](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/empty_lines.rb#L12) configurable, so that people could allow up to two blank lines between method definitions instead of exactly one?

**Comment by bbatsov** [comment]
But if we do what you suggested you would never be certain if 2 lines are there intentionally or not. I'm not saying I'm against your proposal, I'm just saying it might end up skipping some problems this way.

**Comment by alexdowad** [comment]
Agree with @bbatsov.

**Comment by hjdivad** [comment]
@bbatsov yes that's true and unavoidable.  Personally I think the benefit outweighs the cost: aside from being *able* to group methods logically it's good to encourage people to do so as it can help reveal a class with many different concerns.

Personally I tend to add grouping comments (as below) but I have seen this so rarely in other ruby code that I wouldn't describe it as a best or common practice.

```rb
class Account
  ####################
  # Getters

  def balance
  end

  def name
  end

  ####################
  # Mutators

  def deposit!(amount:)
  end
```

My proposal is to let the line offset be configured at either 2 or 3.

Pros:
  1. Able to group methods within a class logically
  2. For projects that don't do this, leave it configured to 2 to avoid false negatives
 
Cons:
  1. If configured to 3, some extra spacing may be missed as a false negative

**Comment by hjdivad** [comment]
@bbatsov thoughts?

**Comment by bbatsov** [comment]
It seems to me like something that's definitely not common practice. :-) I typically avoid supporting those as the benefit doesn't justify the work. Maybe should just disable this cop?

**Comment by hjdivad** [comment]
@bbatsov okie dokie

**Comment by JoshCheek** [comment]
I'll use 2 empty lines to separate methods if the methods have an empty line within them (eg on a test I might use a blank line between given/when/then steps).
